code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -134,6 +134,11 @@ This saves us from having to call add_fingerprint() any time something is put in
 		glasses = null
 		update_inv_glasses()
 	else if (W == head)
+		if(istype(head, /obj/item/clothing/head))
+			var/obj/item/clothing/head/H = head
+			if(H.over_suit == FALSE) // Reset layer changes on removal
+				H.over_suit = TRUE
+				update_inv_head()
```

Comment:
this copypaste is really ugly and I'm not sure why you'd want the layer change to be reset on unequip? no other clothing item in the game does this

also, the inv updates are already done below each of your istypes, you don't need to repeat it

Reply:
This was done because after the head = null (or equivalent further down) line, the update_inv procs cannot tell which item was unequipped (and hence which layer it was on), so they would either need to clear both layers (potentially clearing something else from the alt-layer) or not actually clear the layer.

I took a look at it again though, and changed it so instead the update_inv procs have the item passed in so they can tell which layer to clear.